sunkit_image/granule.py

```diff
@@ -0,0 +1,331 @@
+"""
+This module contains functions that will segment images for granule detection.
+"""
+import logging
+
+import numpy as np
+import scipy.ndimage as sndi
+import skimage
+from sklearn.cluster import KMeans
+
+import sunpy
+import sunpy.map
+
+__all__ = [
+    "segment",
+    "get_threshold",
+    "trim_intergranules",
+    "mark_faculae",
+    "kmeans_segment",
+    "cross_correlation",
+]
+
+
+def segment(data_map, skimage_method, res, mark_dim_centers=False):
+    """
+    Segment optical image of the solar photosphere into tri-value maps
+    with 0 = intergranule, 0.5 = faculae, 1 = granule.
+
+    Parameters
+    ----------
+    data_map : `SunPy map`
+        SunPy map containing data to segment
+    skimage_method : `str`
+        skimage thresholding method, with options 'otsu', 'li', 'isodata',
+        'mean', 'minimum', 'yen', and 'triangle'.
+        Note - depending on input data, one or more of these methods may be
+        signifcantly better or worse than the others. Typically, 'li', 'otsu',
+        'mean', and 'isodata' are good choices, 'yen' and 'triangle' over-
+        identify intergranule material, and 'minimum' overidentifies granules.
+    res : `float`

```

Comment:
I am wondering if this should in fact be a `~astropy.units.quantity.Quantity`. This would have consequences for other parts of the code.

Reply:
Hmm... would do you think it would be easier for the user to input it that way? I am not super farmilier with astropy.units; do you just mean replacing the requirenment that this is in arcseconds with a requirenment that it is an astropy.units.quantity quantity, such that the units are included?

Reply:
Basically it would force a user to pass in a united value, aka 0.13 * u.arcsec/u.pixel.